src/AppBundle/Service/ReportService.php

```diff
@@ -296,4 +324,84 @@ public function findDeleteableReports(Collection $reports)
 
         return $ret;
     }
+
+
+    /**
+     * @param string $select reports|count
+     * @param string $status see Report::STATUS_* constants
+     * @param integer $userId
+     * @param boolean $exclude_submitted
+     * @param string $q search query client firstname/lastname or case number
+     *
+     * @return QueryBuilder
+     */
+    public function getAllReportsQb($select, $status, $userId, $exclude_submitted, $q)
+    {
+        $qb = $this->_em->getRepository(Report::class)
```

Comment:
All repository classes have access to the entity manager. And the reportRepository is already constructed. Hence this method should be in the repository, along with anything else that constructs a query.

Reply:
yes, i stopped refactor and didn't notice there were no dependencies anymore

Reply:
I moved it, pushed